pipeline/backend/kubernetes/pod_test.go

```diff
@@ -351,3 +351,458 @@ func TestFullPod(t *testing.T) {
 	ja := jsonassert.New(t)
 	ja.Assertf(string(podJSON), expected)
 }
+
+func TestFullPodAffinity(t *testing.T) {
```

Comment:
If you want separate test for affinity, then you don't need security context, ports, etc.